coalib/processes/Processing.py

```diff
@@ -267,16 +267,17 @@ def get_file_dict(filename_list, log_printer=None, allow_raw_files=False):
     :return:                Reads the content of each file into a dictionary
                             with filenames as keys.
     """
-    file_dict = {}
+    file_dict = FileDict()
     for filename in filename_list:
         try:
-            with open(filename, 'r',
-                      encoding=detect_encoding(filename)) as _file:
-                file_dict[filename] = tuple(_file.readlines())
+            file_dict[filename] = FileFactory(filename)
+            FileFactory(filename).string
         except UnicodeDecodeError:
             if allow_raw_files:
                 file_dict[filename] = None
                 continue
+            else:
+                del file_dict[filename]
```

Comment:
Deleting the object here makes not sense to me ... the objective of FileFactory was to be able to provide objects which could wrap files which are 1) raw, 2) alternative encodings, 3) etc.

It seems like this is only barely integrated, which is ok I guess at this stage, but you need to add comments where this patch is adding hacks to get the `FileFactory` objects in use, and it seems this `del` seems like it is intentionally reducing the integration of `FileFactory` objects because other parts of core havent been updated yet.

If the 'WIP' was because this wasnt finished yet, unfortunately time has run out, and this patch needs to get merged, so this `del` needs to stay otherwise you'll get more test failures.

Reply:
I needed to evict the file factories when the raw files were not allowed to make the tests pass. I have added the comments explaining the deletion.